owasp-top10-2017-apps/a5/ecommerce-api/app/middlewares/session.go

```diff
@@ -0,0 +1,44 @@
+package middlewares
+
+import (
+	"fmt"
+	"net/http"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/labstack/echo"
+)
+
+// RequireLogin validates the session cookie and sets the username in the context
+func RequireLogin(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		cookie, err := c.Cookie("sessionIDa5")
+		if err != nil {
+			return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": err.Error()})
+		}
+
+		token, err := jwt.Parse(cookie.Value, func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+			}
+			return []byte("secret"), nil
```

Comment:
👀 That was an easter egg! Be careful when hardcoding passwords on code. What do you think of using `os.Getenv("A5-JWT-SECRET")` ?